File: 0106.Construct Binary Tree from Inorder and Postorder Traversal/tree.scala
/**
 * Definition for a binary tree node.
 * class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
 *   var value: Int = _value
 *   var left: TreeNode = _left
 *   var right: TreeNode = _right
 * }
 */
object Solution {
    def buildTree(inorder: Array[Int], postorder: Array[Int]): TreeNode = {
        build(inorder, postorder, 0, inorder.length - 1, 0, postorder.length - 1);
    }
    def build(inorder: Array[Int], postorder: Array[Int], ilow: Int, ihigh: Int, plow: Int, phigh: Int): TreeNode = {
        var ret: TreeNode = null;
        if(ilow == ihigh || plow == phigh)
            ret = new TreeNode(inorder(ilow));
        else if(ilow < ihigh && plow < phigh){
            val rootVal = postorder(phigh);
            ret = new TreeNode(rootVal);
            val pos = findPos(inorder, ilow, ihigh, rootVal);
            ret.left = build(inorder, postorder, ilow, pos - 1, plow, plow + (pos - ilow) - 1);
            ret.right = build(inorder, postorder, pos + 1, ihigh, plow + (pos - ilow), phigh - 1);
        }
        return ret;
    }
    def findPos(v: Array[Int], low: Int, high: Int, target: Int): Int = {
        var i = 0;
        for(i <- low to high){
            if(v(i) == target)
                return i;
        }
        return -1;
    }
}

File: 0106.Construct Binary Tree from Inorder and Postorder Traversal/tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* buildTree(vector<int>& inorder, vector<int>& postorder) {
        return build(inorder, postorder, 0, inorder.size() - 1, 0, postorder.size() - 1);
    }
    
    TreeNode* build(const vector<int>& inorder, const vector<int>& postorder, int ilow, int ihigh, int plow, int phigh){
        //cout << ilow << "," << ihigh << "," << plow << "," << phigh << endl;
        if(ilow > ihigh || plow > phigh)
            return nullptr;
        if(ilow == ihigh)
            return new TreeNode(inorder[ilow]);
        if(plow == phigh)
            return new TreeNode(postorder[plow]);
        int rootVal = postorder[phigh];
        TreeNode *root = new TreeNode(rootVal);
        int pos = findPos(inorder, ilow, ihigh, rootVal);
        root->left = build(inorder, postorder, ilow, pos - 1, plow, plow + (pos - ilow) - 1);
        root->right = build(inorder, postorder, pos + 1, ihigh, plow + (pos - ilow), phigh - 1);
        return root;
    }
    int findPos(const vector<int>& v, int low, int high, int target){
        for(int i = low; i <= high; i++){
            if(v[i] == target)
                return i;
        }
        return -1;
    }
};